src/openfermion/ops/_ising_operator_test.py

```diff
@@ -0,0 +1,221 @@
+                                        numpy.complex128(-1j)])
+def test_imul_scalar(multiplier):
+    loc_op = ((1, 'Z'), (2, 'Z'))
+    qubit_op = IsingOperator(loc_op)
+    qubit_op *= multiplier
+    assert qubit_op.terms[loc_op] == pytest.approx(multiplier)
+
+
+def test_imul_qubit_op():
+    op1 = IsingOperator(((0, 'Z'), (3, 'Z'), (8, 'Z'), (11, 'Z')), 3.j)
+    op2 = IsingOperator(((1, 'Z'), (3, 'Z'), (8, 'Z')), 0.5)
+    op1 *= op2
+    correct_term = ((0, 'Z'), (1, 'Z'), (11, 'Z'))
+    assert len(op1.terms) == 1
+    assert correct_term in op1.terms
+
+
+def test_imul_qubit_op_2():
+    op3 = IsingOperator(((1, 'Z'), (0, 'Z')), -1j)
+    op4 = IsingOperator(((1, 'Z'), (0, 'Z'), (2, 'Z')), -1.5)
+    op3 *= op4
+    op4 *= op3
+    assert ((2, 'Z'),) in op3.terms
+    assert op3.terms[((2, 'Z'),)] == 1.5j
+    assert op4.terms[((0, 'Z'), (1, 'Z'))] == -2.25j
+
+
+def test_imul_bidir():
+    op_a = IsingOperator(((1, 'Z'), (0, 'Z')), -1j)
+    op_b = IsingOperator(((1, 'Z'), (0, 'Z'), (2, 'Z')), -1.5)
+    op_a *= op_b
+    op_b *= op_a
+    assert ((2, 'Z'),) in op_a.terms
+    assert op_a.terms[((2, 'Z'),)] == 1.5j
+    assert ((0, 'Z'), (1, 'Z')) in op_b.terms
+    assert op_b.terms[((0, 'Z'), (1, 'Z'))] == -2.25j
+
+
+def test_imul_bad_multiplier():
+    operator = IsingOperator(((1, 'Z'), (0, 'Z')), -1j)
+    with pytest.raises(TypeError):
+        operator *= "1"
+
+
+def test_mul_by_scalarzero():
+    operator = IsingOperator(((1, 'Z'), (0, 'Z')), -1j) * 0
+    assert ((0, 'Z'), (1, 'Z')) in operator.terms
+    assert operator.terms[((0, 'Z'), (1, 'Z'))] == pytest.approx(0.0)
+
+
+def test_mul_bad_multiplier():
+    operator = IsingOperator(((1, 'Z'), (0, 'Z')), -1j)
+    with pytest.raises(TypeError):
+        operator = operator * "0.5"
+
+
+def test_mul_out_of_place():
+    op1 = IsingOperator(((0, 'Z'), (3, 'Z'), (8, 'Z'), (11, 'Z')), 3.j)
+    op2 = IsingOperator(((1, 'Z'), (3, 'Z'), (8, 'Z')), 0.5)
+    op3 = op1 * op2
+    correct_coefficient = 1.5j
+    correct_term = ((0, 'Z'), (1, 'Z'), (11, 'Z'))
+    assert op1 == IsingOperator(((0, 'Z'), (3, 'Z'), (8, 'Z'), (11, 'Z')), 3.j)
+    assert op2 == IsingOperator(((1, 'Z'), (3, 'Z'), (8, 'Z')), 0.5)
+    assert op3 == IsingOperator(correct_term, correct_coefficient)
+
+
+def test_mul_npfloat64():
+    operator = IsingOperator(((1, 'Z'), (3, 'Z')), 0.5)
+    res = operator * numpy.float64(0.5)
+    assert res == IsingOperator(((1, 'Z'), (3, 'Z')), 0.5 * 0.5)
+
+
+def test_mul_multiple_terms():
+    operator = IsingOperator(((1, 'Z'), (3, 'Z'), (8, 'Z')), 0.5)
+    operator += IsingOperator(((1, 'Z'), (3, 'Z'), (8, 'Z')), 1.2)
+    operator += IsingOperator(((1, 'Z'), (3, 'Z'), (9, 'Z')), 1.4j)
+    res = operator * operator
+    correct = IsingOperator((), 1.7**2 - 1.4**2)
+    correct += IsingOperator(((8, 'Z'), (9, 'Z')),
+                             2j * 1.7 * 1.4)
+    assert res == correct
+
+
+def test_get_operators_empty():
```

Comment:
They shouldn't be in the IsingOperator file, because they just duplicate the QubitOperator ones. If SymbolicOperator becomes an ABC (which I fully support, BTW), then it'll need to be instantiated with a concrete class for testing, so it seems the get_operators tests should just stay in the QubitOperator test file.